junit5/src/test/java/org/jboss/weld/junit5/nested/OuterTestClassAndBeanSameInstanceTest.java

```diff
@@ -0,0 +1,81 @@
+package org.jboss.weld.junit5.nested;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Objects;
+
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.jboss.weld.junit5.auto.EnableAutoWeld;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests that outer test classes don't get instantiated again for {@code @}{@link Produces Produces},
+ * {@code @}{@link Disposes Disposes}, or {@code @}{@link Observes Observes} annotations.
+ * 
+ * Pay attention to with particular instance each outer class member variable reference would refer to
+ * if CDI instantiated another undesired instance of the outer test class besides the one obviously instantiated by JUnit.
+ */
+@EnableAutoWeld
+public class OuterTestClassAndBeanSameInstanceTest {
+
+    interface Bean {
+        default String ping() {
+            // use hashCode to verify same instance in asserts further down thereby ignoring proxies
+            return "ping" + Objects.hashCode(this);
+        }
+    }
+
+    Bean theBean = new Bean() {
+    };
+
+    @Produces
+    Bean produceBean() {
+        return theBean;
+    }
+
+    void disposeBean(@Disposes Bean bean) {
+        assertEquals(theBean.ping(), bean.ping());
+    }
+
+    @Inject @Any
+    Event<Bean> event;
+
+    Bean observedBean;
+
+    void observeBean(@Observes Bean bean) {
+        assertEquals(theBean.ping(), bean.ping());
+        observedBean = theBean;
+    }
+
+    @Nested
```

Comment:
With the changes in this PR also came to add outer test classes to Weld bean classes to deploy automatically. In order to see the test results with the current master branch with two outer test class instances, add

    @AddBeanClasses({OuterTestClassAndBeanSameInstanceTest.class, Bean.class})

That makes this test run with the current master branch without all the other changes of this PR and also failing for the right reason.